bigquery/google/cloud/bigquery/bqdb/cursor.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Cursor for the Google BigQuery DB-API."""
+
+import uuid
+
+from google.cloud.bigquery import job
+from google.cloud.bigquery.bqdb import _helpers
+
+
+class Cursor(object):
+    """Cursor to Google BigQuery.
+
+    :type: :class:`~google.cloud.bigquery.bqdb.Connection`
+    :param connection: A connection to Google BigQuery.
+    """
+    def __init__(self, connection):
+        self.connection = connection
+        self.description = None
+        self.rowcount = -1
+        self._query_results = None
+        self._page_token = None
+        self._has_fetched_all_rows = True
+
+    def close(self):
+        """No-op."""
+        pass
+
+    def _set_description(self, schema):
+        """Set description from schema."""
+        if schema is None:
+            self.description = None
+            return
+
+        desc = []
+        for field in schema:
+            desc.append(tuple([
+                field.name,
+                None,
+                None,
+                None,
+                None,
+                None,
+                field.mode == 'NULLABLE']))
+        self.description = tuple(desc)
+
+    def execute(self, operation):
+        """Prepare and execute a database operation."""
+        self._query_results = None
+        self._page_token = None
+        self._has_fetched_all_rows = False
+        client = self.connection._client
+        job_id = str(uuid.uuid4())
+        # TODO: parameters: if not ``None``, check if ``dict`` or sequence and
```

Comment:
I'd prefer not to merge incomplete implementations here, as it might block us from release or make us release partially implemented features. @dhermes might feel differently, and if he does, then I defer to him.